resin/logs.py

```diff
@@ -18,104 +21,50 @@ class Logs(object):
     """
 
     def __init__(self):
+        self.base_request = BaseRequest()
         self.config = Config()
         self.device = Device()
+        self.settings = Settings()
 
-    def _init_pubnub(func):
-        @wraps(func)
-        def wrapper(self, *args, **kwargs):
-            if not hasattr(self, 'pubnub'):
-                pubnub_key = self.config.get_all()['pubnub']
-                self.pubnub = Pubnub(
-                    publish_key=pubnub_key['publish_key'],
-                    subscribe_key=pubnub_key['subscribe_key']
-                )
-            return func(self, *args, **kwargs)
-        return wrapper
-
-    @_init_pubnub
-    def subscribe(self, uuid, callback, error):
+    def subscribe(self, uuid, callback):
         """
         This function allows subscribing to device logs.
-        There are fields (`m`, `t`, `s`, `c`) in the output which can be unclear. They stand for:
-            m - The log message itself.
-            t - The log timestamp.
-            s - Is this a system message?
-            c - The id of the service which produced this log (or null if the device does not support multiple containers).
 
         Args:
             uuid (str): device uuid.
             callback (function): this callback is called on receiving a message from the channel.
-            error (function): this callback is called on an error event.
-            For more details about callbacks in pubnub subscribe, visit here: https://www.pubnub.com/docs/python/api-reference#subscribe
-
-        Examples:
-            # Define callback and error.
-            >>> def callback(message, channel):
-            ...     print(message)
-            >>> def error(message):
-            ...     print('Error:'+ str(message))
-            >>> Logs.subscribe(uuid=uuid, callback=callback, error=error)
 
         """
 
-        channel = self.get_channel(uuid)
-        self.pubnub.subscribe(channels=channel, callback=callback, error=error)
-
-    @_init_pubnub
-    def history(self, uuid, callback, error):
+        raw_query = 'stream=1'
+        response = self.base_request.request(
+            '/device/v2/{uuid}/logs'.format(uuid=uuid), 'GET', raw_query=raw_query, stream=True,
+            endpoint=self.settings.get('api_endpoint')
+        )
+
+        if (response.status_code == 200):
+            for line in response.iter_lines(chunk_size=None):
+                if line:
+                    entry = json.loads(line)
+                    callback(entry)
```

Comment:
The stream stays open forever, so this function will never return (unless it throws an error). That's pretty different to the previous behaviour, which returned immediately and called the callback asynchronously. I expect it's also fairly inconvenient from an end user point of view. Can we avoid that?

Reply:
@pimterry, this is the same behaviour as the pubnub SDK before. Should I keep the old behaviour or make it consistent with the JS SDK?